File: plugins/plugin-elizacloud/rust/src/models/transcription.rs
#![allow(missing_docs)]

use crate::error::Result;
use crate::providers::ElizaCloudClient;
use crate::types::{ElizaCloudConfig, TranscriptionParams};

pub async fn handle_transcription(
    config: ElizaCloudConfig,
    params: TranscriptionParams,
) -> Result<String> {
    let client = ElizaCloudClient::new(config)?;
    client.transcribe_audio(params).await
}

File: plugins/plugin-elizacloud/python/elizaos_plugin_elizacloud/models/transcription.py
from elizaos_plugin_elizacloud.providers import ElizaCloudClient
from elizaos_plugin_elizacloud.types import ElizaCloudConfig, TranscriptionParams


async def handle_transcription(
    config: ElizaCloudConfig,
    params: TranscriptionParams,
) -> str:
    async with ElizaCloudClient(config) as client:
        return await client.transcribe_audio(params)
